pimd/pim_register.c

```diff
@@ -677,9 +679,12 @@ int pim_register_recv(struct interface *ifp, pim_addr dest_addr,
 			}
 		}
 
-		if ((upstream->sptbit == PIM_UPSTREAM_SPTBIT_TRUE)
-		    || ((SwitchToSptDesiredOnRp(pim, &sg))
-			&& pim_upstream_inherited_olist(pim, upstream) == 0)) {
+		if ((upstream->sptbit == PIM_UPSTREAM_SPTBIT_TRUE) ||
+		    (PIM_UPSTREAM_FLAG_TEST_FHR(upstream->flags) && i_am_rp) ||
```

Comment:
As discussed over slack, please check why upstream->sptbit is not getting set.

Reply:
Yes, I got occupied with different work internally. I will check and update here later.